adapters/smaato/smaato.go

```diff
@@ -298,22 +432,85 @@ func parseImpressionObject(imp *openrtb2.Imp) error {
 		return nil
 	}
 
-	return fmt.Errorf("invalid MediaType. SMAATO only supports Banner and Video. Ignoring ImpID=%s", imp.ID)
+	return &errortypes.BadInput{Message: "Invalid MediaType. Smaato only supports Banner and Video."}
 }
 
-func extractUserExtAttributes(userExt userExt, userCopy *openrtb2.User) {
-	gender := userExt.Data.Gender
-	if gender != "" {
-		userCopy.Gender = gender
+func setImpForAdBreak(imps []openrtb2.Imp) error {
+	adBreakID, err := jsonparser.GetString(imps[0].Ext, "bidder", "adbreakId")
+	if err != nil {
+		return &errortypes.BadInput{Message: "Missing adbreakId parameter."}
 	}
 
-	yob := userExt.Data.Yob
-	if yob != 0 {
-		userCopy.Yob = yob
+	for i := range imps {
+		imps[i].TagID = adBreakID
+		imps[i].Ext = nil
+
+		videoCopy := *(imps[i].Video)
+
+		videoCopy.Sequence = int8(i + 1)
+		videoCopy.Ext, _ = json.Marshal(&videoExt{Context: "adpod"})
+
+		imps[i].Video = &videoCopy
 	}
 
-	keywords := userExt.Data.Keywords
-	if keywords != "" {
-		userCopy.Keywords = keywords
+	return nil
+}
+
+func setBannerDimension(banner *openrtb2.Banner) (*openrtb2.Banner, error) {
+	if banner.W != nil && banner.H != nil {
```

Comment:
Can you please add a JSON test for this condition as well?

Reply:
also created a test for this